lnwallet/channel.go

```diff
@@ -546,6 +547,13 @@ type LightningChannel struct {
 	// way to lookup the original PaymentDescriptor.
 	rHashMap map[PaymentHash][]*PaymentDescriptor
 
+	// [Our|Their]ExpectedBalance reflects the available balance both locally
+	// and remotely. Every time an update happens such as adding or receiving
+	// a HTLC, these values are updated immediately to reflect the available
+	// balance and make sure they don't go below the channel reserve limit.
+	OurExpectedBalance   btcutil.Amount
```

Comment:
As pointed out in my PR-level comment, I think this is the wrong level of abstraction to enforce these limits.